lib/redcord/migration/index.rb

```diff
@@ -5,6 +5,41 @@
 module Redcord::Migration::Index
   extend T::Sig
 
+  sig { params(model: T.class_of(Redcord::Base), index_name: Symbol).void }
+  def add_index(model, index_name)
+    if model.class_variable_get(:@@range_index_attributes).include?(index_name)
+      model.redis.sadd("#{model.model_key}:range_index_attrs", index_name.to_s)
+    elsif model.class_variable_get(:@@index_attributes).include?(index_name)
+      model.redis.sadd("#{model.model_key}:index_attrs", index_name.to_s)
+    else
+      raise(
+        Redcord::AttributeNotIndexed,
+        "#{index_name} is not an indexed attribute.",
+      )
+    end
+
+    # Loop through existing records and build the index
+    model.redis.scan_each(match: "#{model.model_key}:id:*") do |key|
+      index_val = model.redis.hmget(key, index_name.to_s).first
+      id = key.split(':').last.to_i
+
+      # Force refresh on the updated index attribute. Use multi to ensure
+      # consistency incase this migration failed in the middle
+      model.redis.multi
+      model.redis.update_hash(
+        model.model_key,
+        id,
+        {index_name => index_val.empty? ? 'redcord_temp_val' : ''},
```

Comment:
I don't really understand this part. Isn't the code below sufficient? 

`update_hash` also resets the TTL for the record, and I'm not sure we want to do that if we're just indexing it and no client has actually touched it

Reply:
the motivation here is to "force refresh" an attribute so the logic in update_hash can add things to the index structures.

Reply:
Why do you need to call `update_hash` twice? I think it might be better to duplicate or refactor the logic in update_hash so we don't reset the TTL on a record?

Reply:
we have some logic to leave the index structure along unless the value has changed.

> so we don't reset the TTL on a record?

 indeed! we should have a separate script for adding index.